Source/buildbindingpython.go

```diff
@@ -845,16 +846,187 @@ func writeMethod(method ComponentDefinitionMethod, w LanguageWriter, NameSpace s
 		cCheckArguments = "self._handle"
 	}
 	doCheckCall := false
-	for k:=0; k<len(method.Params); k++ {
-		param := method.Params[k]
+
+		// Sort parameters to ensure "in" parameters come before "out" parameters
+	// This creates a copy of the slice to avoid modifying the original
+	sortedParams := make([]ComponentDefinitionParam, len(method.Params))
+	copy(sortedParams, method.Params)
+
+	// Sort by ParamPass: "in" first, then "return", then "out"
+	sort.Slice(sortedParams, func(i, j int) bool {
+		orderMap := map[string]int{"in": 0, "out": 1, "return": 2}
+		return orderMap[sortedParams[i].ParamPass] < orderMap[sortedParams[j].ParamPass]
+	})
+
+	for k := 0; k < len(sortedParams); k++ {
```

Comment:
by looping over the sorted params, you get the intended order of parameters of the methods in Python (in, return, out), but you also break the order of the parameters that are being passed on to the c-ABI-function, see
https://github.com/Autodesk/AutomaticComponentToolkit/pull/228/files#diff-edd85a7b149714f5c67232c1051cf196a96884940a30a598f908e289ed5fed86R897 and 
https://github.com/Autodesk/AutomaticComponentToolkit/pull/228/files#diff-edd85a7b149714f5c67232c1051cf196a96884940a30a598f908e289ed5fed86R1221